arborist/resource_test.go

```diff
@@ -0,0 +1,26 @@
+package arborist
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEncodeDecode(t *testing.T) {
+	inputs := []string{
+		"test",
+		"!@#$%^&*()`\\=+-_'\"<>?,.",
+		"___ooOO00O___",
+		"foo-=-bar-=-baz",
+		"🙃",
```

Comment:
it would be nice to also test a string that contains `_S0`

Reply:
Added in 5899123